assets/js/modules/ads/components/common/WooCommerceRedirectModal.js

```diff
@@ -138,7 +149,10 @@ export default function WooCommerceRedirectModal( {
 	const { dismissNotification } = useDispatch( CORE_NOTIFICATIONS );
 
 	useMount( () => {
-		if ( isGoogleForWooCommerceAdsConnected ) {
+		if (
+			isWooCommerceActivated &&
+			! isAccountLinkedViaGoogleForWoocommerceNoticeDismissed
+		) {
 			dismissNotification( 'account-linked-via-google-for-woocommerce' );
```

Comment:
Let's make a `onMountCallback` prop, and invoke it here
```suggestion
			onMountCallback?.();
```

And then pass `dismissNotification` if not dismissed as the callback of `onMountCallback` in the `SetupMainPAX` component